core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -365,114 +397,88 @@ struct MDRangePolicy : public Kokkos::Impl::PolicyTraits<Properties...> {
   bool impl_tune_tile_size() const { return m_tune_tile_size; }
 
  private:
-  void init() {
-    // Host
-    if (true
+  // Host
+  template <typename ExecutionSpace>
+  void init(const ExecutionSpace&) {
+    const index_type max_threads       = std::numeric_limits<int>::max();
+    const index_type max_tile_size     = 0;
```

Comment:
???

Reply:
The largest tile size was the maximum of the extent and 1. The other execution spaces use a value independent of the extent, though. To make both of these cases work, I decided to treat the case `max_tile_size==0` differently below.